src/ostree/ot-builtin-refs.c

```diff
@@ -70,36 +70,25 @@ static gboolean do_ref (OstreeRepo *repo, const char *refspec_prefix, GCancellab
           g_print ("%s\n", ref);
         }
     }
-  else if(opt_create)
+  else if (opt_create)
     {
-      gboolean old_ref_exists = FALSE;
-      const char *checksum = NULL;
+      g_autofree char *checksum = NULL;
+      g_autofree char *checksum_existing = NULL;
 
-      g_hash_table_iter_init (&hashiter, refs);
-      while (g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue))
+      if (ostree_repo_resolve_rev (repo, opt_create, TRUE, &checksum_existing, error))
```

Comment:
This needs to be flattened, i.e.,

```
if (!ostree_repo_resolve_rev (...))
  goto out;
if (checksum_existing != NULL)
  set_error()
```

Otherwise if an unrelated error occurs (e.g. `EIO`, disk failing), we continue on.